### ISSUE TITLE
FusionViewConcreteDomain3 Failure

### ISSUE THREAD

**Issue by rdspring1** [csarofeen/pytorch#1608]
### 🐛 Describe the bug

NVFuserTest.FusionViewConcreteDomain3_CUDA fails

```
C++ exception with description "index_map.find(root_dom[i]) != index_map.end() INTERNAL ASSERT FAILED at "../torch/csrc/jit/codegen/cuda/index_compute.cpp":2135, please report a bug to PyTorch. Couldn't find root mapping for TV8 dim: 1 id: iS39{12}
Exception raised from getNonGlobalConsumerStridedIndices at ../torch/csrc/jit/codegen/cuda/index_compute.cpp:2135 (most recent call first):
frame #0: c10::Error::Error(c10::SourceLocation, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >) + 0x6c (0x7f7e7ebd1b9c in /home/rds4/pytorch/build/lib/libc10.so)
frame #1: c10::detail::torchCheckFail(char const*, char const*, unsigned int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) + 0xfa (0x7f7e7ebadd9c in /home/rds4/pytorch/build/lib/libc10.so)
frame #2: c10::detail::torchInternalAssertFail(char const*, char const*, unsigned int, char const*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) + 0x53 (0x7f7e7ebcf8b3 in /home/rds4/pytorch/build/lib/libc10.so)
frame #3: <unknown function> + 0x312291a (0x7f7e81d2e91a in /home/rds4/pytorch/build/lib/libtorch_cuda.so)
frame #4: <unknown function> + 0x31236dc (0x7f7e81d2f6dc in /home/rds4/pytorch/build/lib/libtorch_cuda.so)
frame #5: <unknown function> + 0x3123953 (0x7f7e81d2f953 in /home/rds4/pytorch/build/lib/libtorch_cuda.so)
frame #6: torch::jit::fuser::cuda::IndexLowering::handle(torch::jit::fuser::cuda::UnaryOp const*) + 0x44 (0x7f7e81dac0d4 in /home/rds4/pytorch/build/lib/libtorch_cuda.so)
frame #7: torch::jit::fuser::cuda::IndexLowering::handle(torch::jit::fuser::cuda::kir::IfThenElse const*) + 0xd0 (0x7f7e81dabff0 in /home/rds4/pytorch/build/lib/libtorch_cuda.so)
frame #8: torch::jit::fuser::cuda::IndexLowering::handle(torch::jit::fuser::cuda::kir::ForLoop const*) + 0xef (0x7f7e81dae7cf in /home/rds4/pytorch/build/lib/libtorch_cuda.so)
frame #9: torch::jit::fuser::cuda::IndexLowering::handle(torch::jit::fuser::cuda::kir::ForLoop const*) + 0xef (0x7f7e81dae7cf in /home/rds4/pytorch/build/lib/libtorch_cuda.so)
frame #10: torch::jit::fuser::cuda::IndexLowering::handle(torch::jit::fuser::cuda::kir::ForLoop const*) + 0xef (0x7f7e81dae7cf in /home/rds4/pytorch/build/lib/libtorch_cuda.so)
frame #11: torch::jit::fuser::cuda::IndexLowering::handle(torch::jit::fuser::cuda::kir::IfThenElse const*) + 0xd0 (0x7f7e81dabff0 in /home/rds4/pytorch/build/lib/libtorch_cuda.so)
frame #12: torch::jit::fuser::cuda::IndexLowering::handle(torch::jit::fuser::cuda::kir::ForLoop const*) + 0xef (0x7f7e81dae7cf in /home/rds4/pytorch/build/lib/libtorch_cuda.so)
frame #13: torch::jit::fuser::cuda::IndexLowering::handle(torch::jit::fuser::cuda::kir::ForLoop const*) + 0xef (0x7f7e81dae7cf in /home/rds4/pytorch/build/lib/libtorch_cuda.so)
frame #14: torch::jit::fuser::cuda::IndexLowering::generate(std::vector<torch::jit::fuser::cuda::Expr*, std::allocator<torch::jit::fuser::cuda::Expr*> > const&) + 0x2f (0x7f7e81dab3df in /home/rds4/pytorch/build/lib/libtorch_cuda.so)
frame #15: torch::jit::fuser::cuda::GpuLower::lower(torch::jit::fuser::cuda::Fusion*, torch::jit::fuser::cuda::DataType) + 0x1099 (0x7f7e81df1479 in /home/rds4/pytorch/build/lib/libtorch_cuda.so)
frame #16: torch::jit::fuser::cuda::FusionExecutor::compileFusion(torch::jit::fuser::cuda::Fusion*, c10::ArrayRef<c10::IValue> const&, torch::jit::fuser::cuda::LaunchParams const&, torch::jit::fuser::cuda::CompileOptions) + 0xd01 (0x7f7e81cd4a81 in /home/rds4/pytorch/build/lib/libtorch_cuda.so)
frame #17: torch::jit::fuser::cuda::FusionKernelRuntime::runKernelWithInput(c10::ArrayRef<c10::IValue> const&, unsigned long, torch::jit::fuser::cuda::SegmentedGroup*) + 0x73b (0x7f7e81d7fa7b in /home/rds4/pytorch/build/lib/libtorch_cuda.so)
frame #18: torch::jit::fuser::cuda::FusionKernelRuntime::runWithInput(c10::ArrayRef<c10::IValue> const&, unsigned long) + 0x22d (0x7f7e81d8016d in /home/rds4/pytorch/build/lib/libtorch_cuda.so)
frame #19: torch::jit::fuser::cuda::FusionExecutorCache::runFusionWithInputs(c10::ArrayRef<c10::IValue> const&) + 0x3cd (0x7f7e81d83c5d in /home/rds4/pytorch/build/lib/libtorch_cuda.so)
frame #20: torch::jit::NVFuserTest_FusionViewConcreteDomain3_CUDA_Test::TestBody() + 0x4a9 (0x556a9fdeae39 in ./build/bin/test_jit)
frame #21: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) + 0x51 (0x556a9fe33411 in ./build/bin/test_jit)
frame #22: <unknown function> + 0x6f8930 (0x556a9fe26930 in ./build/bin/test_jit)
frame #23: <unknown function> + 0x6f8dc5 (0x556a9fe26dc5 in ./build/bin/test_jit)
frame #24: <unknown function> + 0x6f9521 (0x556a9fe27521 in ./build/bin/test_jit)
frame #25: testing::internal::UnitTestImpl::RunAllTests() + 0x10e9 (0x556a9fe28ca9 in ./build/bin/test_jit)
frame #26: testing::UnitTest::Run() + 0x98 (0x556a9fe291c8 in ./build/bin/test_jit)
frame #27: main + 0xcc (0x556a9f98d83c in ./build/bin/test_jit)
frame #28: __libc_start_main + 0xf3 (0x7f7e7e6f50b3 in /lib/x86_64-linux-gnu/libc.so.6)
frame #29: _start + 0x2e (0x556a9f9d364e in ./build/bin/test_jit)
" thrown in the test body.
```

### Versions

ToT of devel branch

**Comment by naoyam** [comment]
The test seems to break with #1548.

**Comment by naoyam** [comment]
It seems that the test was working just by luck. The recent compute-at-map refactoring changed some ordering, which resulted in this failure, but there's no constraint on the ordering, so the test could have failed previously as well.

**Comment by naoyam** [comment]
Created a fix: #1615
